eras/alonzo/formal-spec/txinfo.tex

```diff
@@ -3,5 +3,284 @@ \section{TxInfo Construction}
+    & \fun{validPlutusdata}~ (\type{P.Constr} ~\wcard ~ds) = \wedge_{d\in ds} (\type{validPlutusdata}~d) \\
+    & \fun{validPlutusdata}~ (\type{P.Map}~ ds) = \wedge_{(x,y)\in ds} ((\type{validPlutusdata}~x) \wedge (\type{validPlutusdata}~x)) \\
+    & \fun{validPlutusdata} ~(\type{P.List}~ ds) =  \wedge_{d\in ds} (\type{validPlutusdata}~d) \\
+    & \fun{validPlutusdata}~ (\type{P.I}~ \wcard) = \True \\
+    & \fun{validPlutusdata}~ (\type{P.B}~ bs) = \fun{length} ~bs~ \leq~ 64 \\
+    & \text{Checks if a Data element is constructed correctly}
+    \nextdef
+    & \fun{validScript} \in \Script \to \Bool \\
+    & \fun{validScript} ~sc = \begin{cases}
+      \True & \text{if $sc \in \ScriptPhOne$} \\
+      \type{P.validateScript} ~sc & \text{if $sc \in \ScriptPhTwo$}
+    \end{cases} \\
+    & \text{Checks if a script is constructed correctly}
+  \end{align*}
+  \caption{Script and Data construction correctness checks}
+  \label{fig:data-script-check}
+\end{figure*}
+
+\subsection{Building Transaction Summary}
+
+The functions in Figure \ref{fig:txinfo-funcs} are needed to build a $\Data$ summary of a transaction.
+The type $\type{P.ScriptContext}$ represents
+the script context. The function $\type{P.toData}$ converts a Plutus-library element
+into a $\Data$ element.
+
+\textbf{$\fun{txInfo}$ Function. }
+$\fun{txInfo}$ summarizes all the necessary transaction and chain state information
+    that needs to be passed to the script interpreter. Below, we specify how to
+    represent relevant transaction data in terms of a Plutus
+    library-defined type. The $\fun{txInfo}$ function builds this representation.
+
+    The $\Language$ argument
+    is required because different languages have different expectations of the
+    format and contents of the $\TxInfo$ summary. The $\EpochInfo$ and $\SystemStart$
+    arguments are needed for translating slot numbers to POSIX time.
+
+    Note that $\fun{txInfo}$ has a $\UTxO$ argument. Even though the full ledger UTxO
+    is passed to it, we define this function in such a way that the only
+    entries in the ledger UTxO map that a Plutus script
+    actually sees via the argument $\fun{txInfo}$ builds are the ones corresponding to the transaction
+    inputs (ie. its realized inputs). This is done in order to maintain the locality of
+    evaluation. For details, see the deterministic script evaluation property~\ref{prop:fixed-inputs}.
+
+\textbf{$\fun{valContext}$ Function. }
+    $\fun{valContext}$ constructs the \emph{validation context}, also referred to as the
+    script context. A validation context is
+    a $\Data$ element which encodes both the summary of the transaction and ledger information
+    (this is supplied by the $\fun{txInfo}$ summarization function), and the script purpose.
+
+\begin{figure}
+  \emph{Plutus Library Types and Functions}
+  %
+  \begin{align*}
+      &\type{P.ScriptContext} ~=~ \type{P.TxInfo} \times \type{P.ScriptPurpose} \\
```

Comment:
This type is unused. If `valContext` was uncurried then it could use that type, but I don't think it should be. So I think this should just be removed.